src/client.rs

```diff
@@ -104,12 +105,43 @@ impl<'a> Client<'a> {
         Client { inner }
     }
 
-    pub fn read_all(&self) -> Vec<Result<ReadByIdxResult, Error<ReadSettingError>>> {
-        const NUM_WORKERS: usize = 5;
+    pub fn read_all(&self) -> (Vec<ReadByIdxResult>, Vec<Error<ReadSettingError>>) {
+        let (_, r) = crossbeam_channel::bounded(0);
+        self.read_all_inner(r)
+    }
+
+    pub fn read_all_timeout(
+        &self,
+        timeout: Duration,
+    ) -> (Vec<ReadByIdxResult>, Vec<Error<ReadSettingError>>) {
+        let (done_s, done_r) = crossbeam_channel::bounded(0);
```

Comment:
I was confused by this in the fileio code too, how does a channel with a upper limit of zero work?  It seems like it would block forever because you can't add anything to it.

Reply:
It's a "handshake"/"rendezvous" channel, which will block senders until a receiver is there to take the value. Same applies in reverse